packages/flutter/lib/src/material/expansion_panel.dart

```diff
@@ -446,7 +446,25 @@ class _ExpansionPanelListState extends State<ExpansionPanelList> {
         context,
         _isChildExpanded(index),
       );
-      final Row header = Row(
+      // ExpandIcon needs to be wrapped with MergeSemantics. If canTapOnHeader
+      // is false, it is possible for headerWidget to contain semantics
+      // boundary. In that case, we wrap the ExpandIcon in expandIconContainer
+      // level. Otherwise, we wrap the entire header as a whole.
+      Widget expandIconContainer = Container(
+        margin: const EdgeInsetsDirectional.only(end: 8.0),
+        child: ExpandIcon(
+          isExpanded: _isChildExpanded(index),
+          padding: const EdgeInsets.all(16.0),
+          onPressed: !child.canTapOnHeader
+              ? (bool isExpanded) => _handlePressed(isExpanded, index)
+              : null,
+        ),
+      );
+      expandIconContainer = !child.canTapOnHeader
+          ? MergeSemantics(
```

Comment:
Why is this MergeSemantics necessary? I wouldn't expect expandIconContainer to contain multiple semantics notes at this point.

Reply:
ExpandIcon does not itself form a semantics node, it need to be wrapped in mergesemantics to form a boundary.

Reply:
MergeSemantics also shouldn't be a semantics boundary? So I am surprised this works.

If you want a semantics boundary, you should use Semantics(container: true, child: ...). But again, I am surprised that you need it here.